### ISSUE TITLE
Android app crash

### ISSUE THREAD

**Issue by StefanoTesla** [nextcloud/android#10929]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

I just run the photo sincronization, the app is unstable, and try to load all the photo tougher (more than 1000)

### Expected behaviour

I think the upload of the photo is not really optimized 
Should be better if he create a set of photo and then try (creating a compressed archive to save traffic?)

### Actual behaviour

When the app is background the upload some time crash, when I restart the application and I go to in uploading section I see more than 1000 upload failed, and he try to upload it again.
At the same time, the app size is very big (something like 20gb).
Seems like the app copy the photo inside to perform the upload, but any error of free space are visible (cause the space is not full).

I give to the app all the permission to run with no save energy setting and the ability to run in background 

### Android version

13

### Device brand and model

Xiaomi 10t pro

### Stock or custom OS?

Stock

### Nextcloud android app version

Last

### Nextcloud server version

24

### Using a reverse proxy?

No

### Cause of error
```java
Exception in thread "WM.task-1" java.lang.OutOfMemoryError: Failed to allocate a 1040 byte allocation with 2917304 free bytes and 2848KB until OOM, target footprint 536870912, growth limit 536870912; giving up on allocation because <1% of heap free after GC.
    at java.io.ObjectInputStream$BlockDataInputStream.<init>(ObjectInputStream.java:2555)
    at java.io.ObjectInputStream.<init>(ObjectInputStream.java:348)
    at androidx.work.Data.fromByteArray(Data.java:433)
    at androidx.work.impl.model.WorkSpecDao_Impl.getAllEligibleWorkSpecsForScheduling(WorkSpecDao_Impl.java:1755)
    at androidx.work.impl.Schedulers.schedule(Schedulers.java:78)
    at androidx.work.impl.WorkerWrapper.onWorkFinished(WorkerWrapper.java:375)
    at androidx.work.impl.WorkerWrapper$2.run(WorkerWrapper.java:331)
    at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
    at java.lang.Thread.run(Thread.java:1012)

```

### App information
* ID: `com.nextcloud.client`
* Version: `30220290`
* Build flavor: `gplay`

### Device information
* Brand: `Xiaomi`
* Device: `apollo`
* Model: `M2007J3SG`
* Id: `SKQ1.211006.001`
* Product: `apollopro_eea`

### Firmware
* SDK: `31`
* Release: `12`
* Incremental: `V13.0.9.0.SJDEUXM`

**Comment by AlvaroBrey** [comment]
Likely will be fixed by #10926

**Comment by AlvaroBrey** [comment]
Duplicate of #10653
